lib/googleauth/compute_engine.rb

```diff
@@ -93,7 +93,7 @@ def fetch_access_token options = {}
           else
             msg = "Unexpected error code #{resp.status}" \
               "#{UNEXPECTED_ERROR_SUFFIX}"
-            raise Signet::AuthorizationError, msg
+            raise Signet::UnexpectedStatusError, msg
```

Comment:
Hi. Sorry, looking at this now. I think we need to distinguish between actual authorization errors and unexpected errors. We don't want to change the behavior (i.e. what exception is raised) if there's a normal authorization error.

What kind of errors are you encountering? If you're getting 5xx errors and you want to return an unexpected status, then please check for that status range explicitly.

Reply:
I encounterd 500 and 403 errors.

Ex
```
Note: Google::Cloud::ErrorReporting is disabled because it failed to authorize with the service. (Unexpected error code 500trying to get security access token from Compute Engine metadata for
```
```
Note: Google::Cloud::ErrorReporting is disabled because it failed to authorize with the service. (Unexpected error code 403trying to get security access token from Compute Engine metadata for
```

`retry_with_error` retries except errors inheritied Signet::AuthorizationError  or Signet::ParseError.

So, How About this?
 
```

          case resp.status
          when 200
            Signet::OAuth2.parse_credentials(resp.body,
                                             resp.headers["content-type"])
          when 403, 500
            msg = "Unexpected error code #{resp.status}" \
              "#{UNEXPECTED_ERROR_SUFFIX}"
            raise Signet::UnexpectedStatusError, msg            
          when 404
            raise Signet::AuthorizationError, NO_METADATA_SERVER_ERROR
          else
            msg = "Unexpected error code #{resp.status}" \
              "#{UNEXPECTED_ERROR_SUFFIX}"
            raise Signet:: AuthorizationError, msg
          end
```

Reply:
ping